io.asgardio.java.oidc.sdk/src/test/java/io/asgardio/java/oidc/sdk/validators/IDTokenValidatorTest.java

```diff
@@ -0,0 +1,292 @@
+                .claim("azp", azpValue)
+                .build();
+
+        SignedJWT idToken = new SignedJWT(new JWSHeader(JWSAlgorithm.RS256), claims);
+        JWSSigner signer = new RSASSASigner(key);
+        idToken.sign(signer);
+
+        IDTokenValidator validator = new IDTokenValidator(config, idToken);
+        IDTokenClaimsSet claimsSet = validator.validate(nonce);
+        List<Audience> audiences = claimsSet.getAudience();
+        audiences.forEach(aud -> assertTrue(trustedAudience.contains(aud.getValue())));
+    }
+
+    @DataProvider(name = "AlgorithmData")
+    public Object[][] algorithmData() throws Exception {
+
+        KeyPairGenerator pairGenRSA = KeyPairGenerator.getInstance("RSA");
+        pairGenRSA.initialize(2048);
+        KeyPair keyPairRSA = pairGenRSA.generateKeyPair();
+
+        RSAKey rsaJWK = new RSAKey.Builder((RSAPublicKey) keyPairRSA.getPublic())
+                .privateKey((RSAPrivateKey) keyPairRSA.getPrivate())
+                .keyID("1")
+                .build();
+
+        ECKey ecJWK = generateECJWK(Curve.P_256);
+
+        return new Object[][]{
+                // algorithm
+                // key
+                {"RS256", (JWK) rsaJWK},
+                {"ES256", (JWK) ecJWK}
+        };
+    }
+
+    @Test(dataProvider = "AlgorithmData")
+    public void testJWSAlgorithm(String signatureAlgorithm, JWK key) throws JOSEException, SSOAgentServerException {
+
+        JWKSet jwkSet = new JWKSet(Collections.singletonList(key));
+
+        Jadler.onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo("/jwksEP")
+                .respond()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody(jwkSet.toJSONObject(true).toJSONString());
+
+        Nonce nonce = new Nonce();
+        JWSAlgorithm jwsAlgorithm = new JWSAlgorithm(signatureAlgorithm);
+        config.setSignatureAlgorithm(jwsAlgorithm);
+        JWTClaimsSet claims = new JWTClaimsSet.Builder()
+                .issuer(config.getIssuer().getValue())
+                .subject("alice")
+                .audience(config.getConsumerKey().getValue())
+                .expirationTime(new Date())
+                .issueTime(new Date())
+                .claim("nonce", nonce.getValue())
+                .build();
+
+        SignedJWT idToken = new SignedJWT(new JWSHeader(jwsAlgorithm), claims);
+        JWSSigner signer;
+        if (key instanceof RSAKey) {
+            signer = new RSASSASigner((RSAKey) key);
+        } else {
+            signer = new ECDSASigner((ECKey) key);
+        }
+        idToken.sign(signer);
+
+        IDTokenValidator validator = new IDTokenValidator(config, idToken);
+        IDTokenClaimsSet claimsSet = validator.validate(nonce);
+        assertEquals(claimsSet.getNonce(), nonce);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+        closeJadler();
+    }
+}
```

Comment:
Add a newline at the end

Reply:
Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/12/commits/8935b73e06c06ac1e1f216c89c0b19e175e1c196.